File: 프로그래머스/unrated/136798. 기사단원의 무기/기사단원의 무기.kt
class Solution {
    fun solution(number: Int, limit: Int, power: Int): Int {
        val list = mutableListOf<Int>()

        for(i in 1..number) {
            list.add(measure(i))
        }

        val notOverLimit = list.filter { it <= limit }.sum()
        val overLimit = list.count { it > limit } * power

        return notOverLimit + overLimit
    }

    fun measure(num: Int): Int {
        var count = 0
        for(i in 1..num/2) {
            if(num % i == 0) {
                count++
            }
        }
        return ++count
    }
}

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.swift
import Foundation

func solution(_ number:Int, _ limit:Int, _ power:Int) -> Int {
    var answer = 0
    
    for i in 1...number {
        var count = getMeasureCount(i)
        answer += (count > limit) ? power : count
    }
    
    return answer
}

func getMeasureCount(_ number: Int) -> Int {
    var measure = Array<Int>()
    
    for i in 1...Int(sqrt(Double(number))) {
        if number % i == 0 {
            measure.append(i)
        }
    }
    
    for i in measure {
        if !measure.contains(number/i) {
            measure.append(number/i)
        }
    }
    
    return measure.count
}